torch/csrc/jit/tensorexpr/loopnest.cpp

```diff
@@ -1911,6 +1911,58 @@ std::vector<For*> LoopNest::reorder(
   return result;
 }
 
+std::vector<For*> LoopNest::tile(For* x, For* y, int x_factor, int y_factor) {
+  auto parent = dynamic_cast<Block*>(x->get_parent());
+  if (parent == nullptr) {
+    throw malformed_input("parent of the loops must be a Block");
+  }
+  if (!areLoopsPerfectlyNested({x, y})) {
+    throw malformed_input("two loops must be perfectly nested");
+  }
+
+  // Split x, y axes by x_factor and y_factor
+  For *yi, *ytail;
+  splitWithTail(y, y_factor, &yi, &ytail);
+  For *xi, *xtail;
+  splitWithTail(x, x_factor, &xi, &xtail);
+
+  // Distribute xi over yo and ytail so we can manipulate the loop order of {xo,
+  // xi, yo, yi}
+  auto loops = distributeLoop(xi);
+
+  // For {xo, xi, yo, yi}, reorder the axes to be xo, yo, xi, yi
+  xi = loops.front();
+  For* yo = dynamic_cast<For*>(xi->body()->stmts().front());
+  CHECK(yo);
+  reorderAxis(xi, yo);
```

Comment:
Since this loopnest is expected to be perfectly-nested, you could use `reorder` API instead here.

Reply:
The loopnest is not perfectly-nested because `xo` has two nested loops inside, i.e., `{xi, yo, yi}` and `{xi, ytail}` so that's why we don't need to fuse any loops afterwards. In summary, we can either use `reorderAxis` in this way or use `reorder` in the way of 1)`distribute` `xo` over the two loops then 2) `reorder` {xo, xi, yo, yi} and finally 3) fuse the two distributed loops. Both are fine but the latter one seems a bit tedious to me.

Reply:
I understand `xo` has two nested loops inside. But since you are only reordering `xi` and `yo`, and they are perfectly nested, you can still use `reorder` API. That API has simpler logic and also performs the transformation in-place. So, you can avoid looking up the loops again below.